File: scrimage-filters/src/main/scala/com/sksamuel/scrimage/filter/RGBFilter.scala
package com.sksamuel.scrimage.filter

import com.sksamuel.scrimage.BufferedOpFilter

class RGBFilter(r: Double = 0, g: Double = 0, b: Double = 0) extends BufferedOpFilter {
  require(r <= 1)
  require(g <= 1)
  require(b <= 1)
  val op = new thirdparty.jhlabs.image.RGBAdjustFilter()
  op.setBFactor(b.toFloat)
  op.setRFactor(r.toFloat)
  op.setGFactor(g.toFloat)
}

object RGBFilter {
  def apply(r: Double = 0, g: Double = 0, b: Double = 0): RGBFilter = new RGBFilter(r, g, b)
}

File: scrimage-filters/src/main/java/com/sksamuel/scrimage/filter/RGBFilter.java
package com.sksamuel.scrimage.filter;

import com.sksamuel.scrimage.BufferedOpFilter;
import thirdparty.jhlabs.image.RGBAdjustFilter;

import java.awt.image.BufferedImageOp;

public class RGBFilter extends BufferedOpFilter {
    //  require(r <= 1)
//  require(g <= 1)
//  require(b <= 1)

    private final float r;
    private final float g;
    private final float b;

    public RGBFilter(float r, float g, float b) {
        this.r = r;
        this.g = g;
        this.b = b;
    }

    public RGBFilter() {
        this(0, 0, 0);
    }

    @Override
    public BufferedImageOp op() {
        RGBAdjustFilter op = new RGBAdjustFilter();
        op.setBFactor(b);
        op.setRFactor(r);
        op.setGFactor(g);
        return op;
    }

}
